src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,520 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 12 June 2020
+
+@author: Anders Muskens
+
+Copyright © 2020 Anders Muskens, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+from __future__ import division
+
+from concurrent.futures import CancelledError, TimeoutError
+import logging
+import fcntl
+import glob
+from odemis import model
+from odemis.model import HwError
+from odemis.util import to_str_escape
+import os
+import re
+import serial
+import threading
+import time
+from odemis.model._vattributes import FloatVA, IntEnumerated
+from _ast import Pass
+
+KEYPAD_UNLOCK = 0
+POLL_INTERVAL = 0.5  # interval to poll new temperature
+
+LAKESHORE_STATUS_BYTE = {
+    128: "Power on",
+    32: "Command Error",
+    16: "Execution error",
+    4: "Query Error",
+    1: "Operation complete"
+    }
+
+POWER_ON = 128
+COMMAND_ERROR = 32
+EXECUTION_ERROR = 16
+QUERY_ERROR = 4
+OPC = 1
+
+
+class LakeshoreError(model.HwError):
+    """
+    Exception used to indicate a problem reported by the device.
+    """
+
+    def __init__(self, msg, byte=None):
+        self.byte = byte
+        model.HwError.__init__(self, msg + " Status byte: %x" % (self.byte,))
+
+
+class Lakeshore(model.HwComponent):
+
+    def __init__(self, name, role, port, sensor_input='b', output_channel=2, **kwargs):
+        """
+        A driver for the Lakeshore 310 temperature controller.
+
+        name: (str)
+        role: (str)
+        port: (str) port name. Can be a pattern, in which case all the ports
+          fitting the pattern will be tried.
+          Use /dev/fake for a simulator
+        sensor_input (str): The sensor input to use, typically 'a' or 'b'
+        output_channel: (int): The channel output to control, typically 1 or 2
+        """
+
+        # Connect to serial port
+        self._ser_access = threading.Lock()
+        self._serial = None
+        self._file = None
+        self._sensor_input = sensor_input
+        self._output_channel = output_channel
+        self._port, self._version = self._findDevice(port)  # sets ._serial and ._file
+        logging.info("Found Lakeshore 310 device on port %s, Ver: %s",
+                     self._port, self._version)
+
+        # Clear errors at start
+        try:
+            self.checkError()
+        except LakeshoreError:
+            pass
+
+        manufacturer, md, serial = self.GetIdentifier()
+
+        self._hwVersion = str(md)
+        # self._swVersion = self._version.decode(serial)
```

Comment:
You can use the same as the other drivers, which just show the linux kernel driver version:
```python
        driver_name = driver.getSerialDriver(self._port)
        self._swVersion = "serial driver: %s" % (driver_name,)
```